src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeSynchronizer.cs

```diff
@@ -1,113 +1,58 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
-using Nethermind.Blockchain;
-using Nethermind.Blockchain.Receipts;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac;
+using Autofac.Features.AttributeFilters;
 using Nethermind.Blockchain.Synchronization;
-using Nethermind.Config;
-using Nethermind.Consensus;
-using Nethermind.Core.Specs;
-using Nethermind.Db;
+using Nethermind.Core;
 using Nethermind.Logging;
-using Nethermind.Merge.Plugin.InvalidChainTracker;
-using Nethermind.Specs.ChainSpecStyle;
-using Nethermind.State;
-using Nethermind.Stats;
 using Nethermind.Synchronization;
 using Nethermind.Synchronization.Blocks;
 using Nethermind.Synchronization.FastBlocks;
 using Nethermind.Synchronization.ParallelSync;
-using Nethermind.Synchronization.Peers;
-using Nethermind.Trie;
-using Nethermind.Trie.Pruning;
 
 namespace Nethermind.Merge.Plugin.Synchronization;
 
-public class MergeSynchronizer : Synchronizer
+public class MergeSynchronizer(
+    [KeyFilter(nameof(BeaconHeadersSyncFeed))] SyncFeedComponent<HeadersSyncBatch> beaconHeaderComponent,
+    ISyncConfig syncConfig,
+    Synchronizer baseSynchronizer,
```

Comment:
```suggestion
    ISynchronizer baseSynchronizer,
``` 

?

Reply:
Need the `WireFeedWithModeSelector` which does not fit nicely into `ISynchronizer`. Alternatively, could copy the whole thing, which is also, not very nice.